### PR TITLE
Fix nodeport repair for ESIPP services

### LINKED ISSUE #54885: Service Nodeport and Healthcheck Nodeport Collision
I came across a cluster where it has 2 services with port collision. 

One service looks like this:
```
type: LoadBalancer
externalTrafficPolicy: Local
healthCheckNodePort: ${Collided Port}
```

The other looks like:
```
...
  annotation: cloud.google.com/load-balancer-type=Internal
Spec: 
  type: LoadBalancer
  externalTrafficPolicy: Cluster
  ports
  - nodeport: ${Collided Port}
    ....
```

I checked the logic that allocates nodeport and healthcheck nodeport. They went thru the same node port allocator. Port allocation mainly happens in `initNodePorts`, `updateNodePorts` and `allocateHealthCheckNodePort`. The file in question is https://github.com/kubernetes/kubernetes/blob/release-1.8/pkg/registry/core/service/rest.go

### PR DESCRIPTION
**Pull request by nicksardo**
**What this PR does / why we need it**:
The nodeport allocation repair controller does not scrape the `Service.Spec.healthCheckNodePort` value and would remove the allocation from memory and etcd after 10 minutes.  This opens the door for other services to use the same nodeport and cause collisions. 

**Which issue(s) this PR fixes**:
Fixes #54885
Similar to #64349

**Release note**:
```release-note
Fix issue of colliding nodePorts when the cluster has services with externalTrafficPolicy=Local
```

### CODE REVIEW

File: pkg/registry/core/service/portallocator/controller/repair.go [hunk 1]
```diff
@@ -185,3 +184,19 @@ func (c *Repair) runOnce() error {
 	}
 	return nil
 }
+
+func collectServiceNodePorts(service *api.Service) []int {
```
[the remark this answers is not in the document]
Comment (nicksardo) [reply pkg/registry/core/service/portallocator/controller/repair.go]: The existing func is used in numerous places which expect it to only handle the port nodeports - not the healthcheck nodeport.

[the remark this answers is not in the document]
Comment (MrHohn) [reply pkg/registry/core/service/portallocator/controller/repair.go]: I see, thanks for clarifying.

**Review by MrHohn — APPROVED** [review]
/lgtm

**Pull request closed without merging**